File: pyg-groupck/pyg-pay/pyg-pay-interface/src/main/java/com/pyg/pay/service/PayService.java
package com.pyg.pay.service;

import java.util.Map;

/**
 * Created by on 2018/9/4.
 */
public interface PayService {
    /**
     * 需求：生成二维码
     * @param userId
     * @return Map
     */
    public Map createQrCode(String userId);

    /**
     * 需求：查询支付二维码状态，查询是否支付成功
     * @param out_trade_no
     * @return
     */
    public Map queryStatus(String out_trade_no);
}

File: pyg-groupck/pyg-pay-web/src/main/webapp/js/service/payService.js
//服务层
app.service('payService', function ($http) {

    //生成二维码
    this.createQrCode = function () {
        return $http.get('../pay/createQrCode');
    };

    //监控二维码支付状态
    this.queryStatus = function (out_trade_no) {
        return $http.get('../pay/queryStatus/'+out_trade_no);
    }

});
